### ISSUE TITLE
NTFS directory junctions are followed

### ISSUE THREAD

**Issue by mattpalermo** [syncthing/syncthing#7422]
Syncthing version: v1.14.0
Windows 10 version: 1909 & 2004
Synctrayzor version: v1.1.27

# NTFS junctions are followed

It seems that Syncthing on Windows 10 will follow NTFS directory junctions as if they were normal directories. This is different from the behaviour for NTFS sym-links which are not followed. Unless I am mistaken, this seems like a bug rather than intended behaviour?

# How to reproduce

Have two windows 10 machines sharing a folder `link-test` with Syncthing. At machine 1, In the `link-test` folder, run the following in command prompt:
```
mkdir dir
mklink /j junction
cd dir
echo hello world > hello.txt
```

At this point, at machine 2 we will get `dir`, `junction` and `dir\hello.txt`. Seeing `junction` here is different to the behaviour with sym-links.
At machine 1, make Syncthing rescan `link-test`, and at machine 2 we will get `junction\hello.txt`. Obviously this is also different from sym-link behaviour.

We can also try to make sync-thing sync an infinitely recursive junction, but it will detect the recursion after only a single iteration. It will report this as a failed item with the reason: "infinite filesystem recursion detected".

# What this affects

`npm link` seems to create junctions rather than sym-links, at least by default. I will look into seeing if it can use sym-links.

**Comment by mattpalermo** [comment]
For anyone whose lands here looking for a solution, my current solution is to simply add the particular part of `node_modules` which is causing problems to `.stignore` on any machine which will run `npm link` (or `npm install` with a project with file dependencies).

**Comment by tomasz1986** [comment]
Do you have `junctionsAsDirs` enabled in the Advanced Configuration for the folder in question? Junctions should be ignored if the setting is disabled, and followed only if it is enabled. The option itself should be disabled by default. It may be enabled for your folder if you have been using the same Syncthing installation for a while, as it used to enabled by default when first implemented, and those settings will remain for the same folder.

Basically, if possible, please try to create a new folder and then reproduce the issue there (without enabling `junctionsAsDirs`).

**Comment by mattpalermo** [comment]
Thankyou @tomasz1986, That was indeed the problem. I know now to test with new folders from now on. I have only the one shared folder that I have been using for a while. It just syncs the files that I'm currently working on between my phone and laptops.
